### ISSUE TITLE
The reproduced results on the odinw13 dataset are lower than those reported.

### ISSUE THREAD

**Issue by HuaijieWang** [chiara-cap/DitHub#3]
**The results on odinw13 is lower than those reported in parpar (table.1). The specific result is：** 
full: coco: **47.95** odinw13: **57.33** (report: coco: **47.01** odinw13: **62.19**)
10-shots: coco: **47.99** odinw13: **50.26** (report: coco: **46.54** odinw13: **54.43**)
5-shots: coco: **48.03** odinw13: **50.80** (report: coco: **45.86** odinw13: **52.85**)
1-shots: coco: **47.96** odinw13: **48.28** (report: coco: **46.40** odinw13: **49.19**)

**My command is：**
# 1shot
CUDA_VISIBLE_DEVICES=1 python -u main.py --config-file test/test_odinw13_1shot --model-config-file groundingdino/config/GroundingDINO_SwinT_OGC_dt_dithub.py --model-checkpoint-path groundingdino_swint_ogc.pth --output-dir /output/dithub_output_1shot --seed 3 --num-gpus 1 --shuffle-tasks --dithub

# 5shot
CUDA_VISIBLE_DEVICES=6 python -u main.py --config-file test/test_odinw13_5shot --model-config-file groundingdino/config/GroundingDINO_SwinT_OGC_dt_dithub.py --model-checkpoint-path groundingdino_swint_ogc.pth --output-dir /output/dithub_output_5shot --seed 3 --num-gpus 1 --shuffle-tasks --dithub

# 10shot
CUDA_VISIBLE_DEVICES=7 python -u main.py --config-file test/test_odinw13_10shot --model-config-file groundingdino/config/GroundingDINO_SwinT_OGC_dt_dithub.py --model-checkpoint-path groundingdino_swint_ogc.pth --output-dir /output/dithub_output_10shot --seed 3 --num-gpus 1 --shuffle-tasks --dithub

# full
CUDA_VISIBLE_DEVICES=1 python -u main.py --config-file test/test_odinw13 --model-config-file groundingdino/config/GroundingDINO_SwinT_OGC_dt_dithub.py --model-checkpoint-path groundingdino_swint_ogc.pth --output-dir /output/dithub_output_full --seed 3 --num-gpus 1 --shuffle-tasks --dithub

**The output details for each setting are：**
1. full:
AP results: {'test/test_odinw13/for_train/test_ShellfishOpenImages.py': 38.608423644708026, 'test/test_odinw13/for_train/test_NorthAmericaMushrooms.py': 34.81777408519805, 
'test/test_odinw13/for_train/test_Packages.py': 59.730473047304734, 
'test/test_odinw13/for_train/test_AerialMaritimeDrone_tiled.py': 31.85731565409852, 
'test/test_odinw13/for_train/test_pistols.py': 67.18609215393805, 
'test/test_odinw13/for_train/test_VehiclesOpenImages.py': 65.9214259942647, 
'test/test_odinw13/for_train/test_Raccoon.py': 71.92081183666266, 
'test/test_odinw13/for_train/test_aquarium.py': 36.25157674433115, 
'test/test_odinw13/for_train/test_thermalDogsAndPeople.py': 78.98642292962957, 
'test/test_odinw13/for_train/test_CottontailRabbits.py': 72.46643211993546, 
'test/test_odinw13/for_train/test_pothole.py': 46.26266845643377, 
'test/test_odinw13/for_train/test_PascalVoc.py': 69.52606654333255, 
'test/test_odinw13/for_train/test_Egohands_generic.py': 71.72382910447645, 
'test/test_odinw13/test_zero_shot_coco.py': 47.95167731796596}
[09/30 01:08:26 detectron2]: average AP: 57.32763940879337
[09/30 01:08:26 detectron2]: AP on COCO: 47.95167731796596

2. 10-shots
AP results: {'test/test_odinw13_10shot/for_train/test_ShellfishOpenImages.py': 32.312438013735516, 'test/test_odinw13_10shot/for_train/test_NorthAmericaMushrooms.py': 32.93782635840036, 
'test/test_odinw13_10shot/for_train/test_Packages.py': 62.17821782178218, 
'test/test_odinw13_10shot/for_train/test_AerialMaritimeDrone_tiled.py': 19.601392792816654, 
'test/test_odinw13_10shot/for_train/test_pistols.py': 66.56380984156417, 
'test/test_odinw13_10shot/for_train/test_VehiclesOpenImages.py': 58.26992225765571, 
'test/test_odinw13_10shot/for_train/test_Raccoon.py': 65.66454252349708, 
'test/test_odinw13_10shot/for_train/test_aquarium.py': 21.650648259965582, 
'test/test_odinw13_10shot/for_train/test_thermalDogsAndPeople.py': 71.53385708468805, 'test/test_odinw13_10shot/for_train/test_CottontailRabbits.py': 67.92574257425744, 
'test/test_odinw13_10shot/for_train/test_pothole.py': 28.55883802192366, 
'test/test_odinw13_10shot/for_train/test_PascalVoc.py': 60.77749607890034, 
'test/test_odinw13_10shot/for_train/test_Egohands_generic.py': 65.43514371763678, 
'test/test_odinw13_10shot/test_zero_shot_coco.py': 47.985852536157296}
[09/30 15:40:41 detectron2]: average AP: 50.26229810360179
[09/30 15:40:41 detectron2]: AP on COCO: 47.985852536157296

3. 5-shots
AP results: {'test/test_odinw13_5shot/for_train/test_ShellfishOpenImages.py': 35.235979823840445, 'test/test_odinw13_5shot/for_train/test_NorthAmericaMushrooms.py': 34.01137677135381,
'test/test_odinw13_5shot/for_train/test_Packages.py': 59.7029702970297, 
'test/test_odinw13_5shot/for_train/test_AerialMaritimeDrone_tiled.py': 19.576289885220575, 
'test/test_odinw13_5shot/for_train/test_pistols.py': 66.33228473697281, 
'test/test_odinw13_5shot/for_train/test_VehiclesOpenImages.py': 58.11120148391525, 
'test/test_odinw13_5shot/for_train/test_Raccoon.py': 67.38591897561534, 
'test/test_odinw13_5shot/for_train/test_aquarium.py': 21.92497878936971, 
'test/test_odinw13_5shot/for_train/test_thermalDogsAndPeople.py': 73.34115366759474, 'test/test_odinw13_5shot/for_train/test_CottontailRabbits.py': 69.0619248584853, 
'test/test_odinw13_5shot/for_train/test_pothole.py': 28.649046641942988, 
'test/test_odinw13_5shot/for_train/test_PascalVoc.py': 60.9664378449798, 
'test/test_odinw13_5shot/for_train/test_Egohands_generic.py': 66.14271589844066, 
'test/test_odinw13_5shot/test_zero_shot_coco.py': 48.02716971671475}
[09/30 15:41:06 detectron2]: average AP: 50.80325228267394
[09/30 15:41:06 detectron2]: AP on COCO: 48.02716971671475

4. 1-shots
AP results: {'test/test_odinw13_1shot/for_train/test_ShellfishOpenImages.py': 33.71148293034944, 'test/test_odinw13_1shot/for_train/test_NorthAmericaMushrooms.py': 30.80830994271222, 
'test/test_odinw13_1shot/for_train/test_Packages.py': 56.27062706270628, 
'test/test_odinw13_1shot/for_train/test_AerialMaritimeDrone_tiled.py': 19.402655083527925, 
'test/test_odinw13_1shot/for_train/test_pistols.py': 65.70905122187052, 
'test/test_odinw13_1shot/for_train/test_VehiclesOpenImages.py': 60.334975467231914, 
'test/test_odinw13_1shot/for_train/test_Raccoon.py': 62.92167526317317, 
'test/test_odinw13_1shot/for_train/test_aquarium.py': 20.807411057458943, 
'test/test_odinw13_1shot/for_train/test_thermalDogsAndPeople.py': 72.02072134671863, 'test/test_odinw13_1shot/for_train/test_CottontailRabbits.py': 65.74778192104925, 
'test/test_odinw13_1shot/for_train/test_pothole.py': 23.282128172850438, 
'test/test_odinw13_1shot/for_train/test_PascalVoc.py': 57.353766663333175, 
'test/test_odinw13_1shot/for_train/test_Egohands_generic.py': 59.219788432068945, 
'test/test_odinw13_1shot/test_zero_shot_coco.py': 47.96183198580734}
[09/29 14:10:05] detectron2 INFO: average AP: 48.27618265885007
[09/29 14:10:05] detectron2 INFO: AP on COCO: 47.96183198580734

**All code changes are:**
1. main.py, in line 247: "last_lora = torch.load(Path(output_dir, 'last_lora.pth'))" -> "last_lora = torch.load(Path(output_dir, 'last_lora.pth'), weights_only=False, map_location='cpu')"
2. groundingdino/util/get_tokenlizer.py, "tokenizer = AutoTokenizer.from_pretrained(text_encoder_type)" -> "
    if os.path.isdir(local_dir) and os.listdir(local_dir):
        tokenizer = AutoTokenizer.from_pretrained(local_dir, local_files_only=True)
    else:
        tokenizer = AutoTokenizer.from_pretrained(text_encoder_type)
        tokenizer.save_pretrained(local_dir)"
3. groundingdino/util/get_tokenlizer.py, "return BertModel.from_pretrained(text_encoder_type)" -> "
        if os.path.isdir(local_dir) and os.listdir(local_dir):
            model = BertModel.from_pretrained(local_dir, local_files_only=True)
        else:
            model = BertModel.from_pretrained(text_encoder_type)
            model.save_pretrained(local_dir)  
        return model
"
It seems that the number of training samples has little effect on zero-shot performance, and it is generally higher than the reported metrics. I think it might be an issue with my reproduction. I wonder if the authors have encountered a similar situation. I would be very grateful if the authors could provide some guidance.

**Comment by chiara-cap** [comment]
Hello, and thank you for sharing your detailed results!

This appears to be the same problem we discussed in a previous issue: #2 . You can find there the explanation and the corrected parameters that resolve the discrepancy.

Please let us know if applying the fix works on your side as well. Thanks again for helping us improve the repo!

**Comment by HuaijieWang** [comment]
> Hello, and thank you for sharing your detailed results!
> 
> This appears to be the same problem we discussed in a previous issue: [#2](https://github.com/chiara-cap/DitHub/issues/2) . You can find there the explanation and the corrected parameters that resolve the discrepancy.
> 
> Please let us know if applying the fix works on your side as well. Thanks again for helping us improve the repo!

Thank you for providing the new running code. Following this instruction, my results are as follows：
full: coco: **47.55** odinw13: **60.82** (report: coco: **47.01** odinw13: **62.19**)
10-shots: coco: **46.15** odinw13: **55.90** (report: coco: **46.54** odinw13: **54.43**)
5-shots: coco: **45.67** odinw13: **54.83** (report: coco: **45.86** odinw13: **52.85**)
1-shots: coco: **46.84** odinw13: **50.94** (report: coco: **46.40** odinw13: **49.19**)
Judging from the results, there are still some differences under the full settings (coco's results are on the high side, and odinw13's results are on the low side). In terms of code changes, I changed the online download of bert's tokenlizer and model to only download it once, and then read it locally. In the configuration environment,  I put the readme file under 
```
cd Deformable-DETR/models/ops
sh ./make.sh
```
This code is run after the following preprocessing:
```
cd Deformable-DETR/models/ops/src/cuda

cp ms_deform_attn_cuda.cu ms_deform_attn_cuda.cu.bak

sed -i 's/AT_DISPATCH_FLOATING_TYPES(value.type(),/AT_DISPATCH_FLOATING_TYPES(value.scalar_type(),/g' ms_deform_attn_cuda.cu

sed -i 's/\.data<\([^>]*\)>()/.data_ptr<\1>()/g' ms_deform_attn_cuda.cu

sed -i 's/\([A-Za-z_][A-Za-z0-9_]*\)\.type()\.is_cuda()/\1.is_cuda()/g' ms_deform_attn_cuda.cu
```
This change is due to environmental compatibility issues.
I ran it three times under the full setting, and the results were all around this. I don’t know why, which makes me a little confused.

**Comment by chiara-cap** [comment]
Hi, and thank you again for the detailed follow-up!

We’ve confirmed that another user was able to reproduce the reported results with the corrected hyperparameters (#2 ), obtaining an average AP of 62.36 and a COCO AP of 47.44, which matches our reference numbers.

Given this, we suspect that the discrepancy you are seeing might be related to your local environment (e.g., library versions, CUDA/torch build, or tokenizer/model caching).

To narrow this down, could you please try running only the evaluation step using our released checkpoint — without retraining — and report the results?
You can do this by running:

```
python -u main.py \
  --config-file test/test_odinw13 \
  --model-config-file groundingdino/config/GroundingDINO_SwinT_OGC_dt_dithub.py \
  --model-checkpoint-path model_final.pth \
  --output-dir /output/dithub_eval_only \
  --seed 3 \
  --num-gpus 1 \
  --shuffle-tasks \
  --dithub \
  --eval-only
```

If the evaluation results match the paper, that would confirm the issue is isolated to the training stage. Otherwise, we can help debug potential environment-related causes (e.g., tokenizer loading, Detectron2/CUDA versions).

**Comment by HuaijieWang** [comment]
> Hi, and thank you again for the detailed follow-up!
> 
> We’ve confirmed that another user was able to reproduce the reported results with the corrected hyperparameters ([#2](https://github.com/chiara-cap/DitHub/issues/2) ), obtaining an average AP of 62.36 and a COCO AP of 47.44, which matches our reference numbers.
> 
> Given this, we suspect that the discrepancy you are seeing might be related to your local environment (e.g., library versions, CUDA/torch build, or tokenizer/model caching).
> 
> To narrow this down, could you please try running only the evaluation step using our released checkpoint — without retraining — and report the results? You can do this by running:
> 
> ```
> python -u main.py \
>   --config-file test/test_odinw13 \
>   --model-config-file groundingdino/config/GroundingDINO_SwinT_OGC_dt_dithub.py \
>   --model-checkpoint-path model_final.pth \
>   --output-dir /output/dithub_eval_only \
>   --seed 3 \
>   --num-gpus 1 \
>   --shuffle-tasks \
>   --dithub \
>   --eval-only
> ```
> 
> If the evaluation results match the paper, that would confirm the issue is isolated to the training stage. Otherwise, we can help debug potential environment-related causes (e.g., tokenizer loading, Detectron2/CUDA versions).

Thank you for your answer. I also think that the results are slightly different due to the environment problem. Due to server problems, in order to run the code, we can only rely on the current environment. 
Thank you again for your answer and your excellent work.
